### PR TITLE
Call diskparts to discover partitions on startup (terminal and CPU)

### PR DESCRIPTION
**Pull request by gmacd**
- Also update diskparts according to a change suggested by Erik Quanstrom in 9fans to correctly discover partitions on disks with no plan9 partition.
- Add both CPU and terminal options to syslinux menu

Signed-off-by: Graham MacDonald <grahamamacdonald@gmail.com>#

### CODE REVIEW

**Comment by elbing** [comment]
I still think Harvey doesn't need swap.

File: rc/bin/cpurc [hunk 1]
```diff
@@ -12,6 +12,15 @@ sysname=`{cat /dev/sysname}
 # parallelism for mk
 NPROC = `{wc -l </dev/sysstat}
 
+# set up any partitions
+diskparts
+
+# start up local swapping
```
Comment (rminnich) [inline rc/bin/cpurc]: no. take this out.

Comment (rminnich) [reply rc/bin/cpurc]: it will fail anyway, just don't create swap partitions, but ... it has not worked in years.

File: rc/bin/termrc [hunk 2]
```diff
@@ -12,22 +12,14 @@ mntgen -s slashn && chmod 666 /srv/slashn
...
-	if (! ~ $#disk 0)
-		swap $disk(1) >/dev/null >[2=1]
-	rm /env/disk
-}
-
-# END OF ONLY LOCAL
+# set up any partitions
+diskparts
+
+# start up local swapping
```
Comment (rminnich) [inline rc/bin/termrc]: remove

File: util/syslinux-bios/syslinux/syslinux.cfg [hunk 3]
```diff
@@ -1,18 +1,27 @@
...
 LABEL 0
 TEXT HELP
-Boot Harvey kernel
+Boot Harvey kernel as a CPU node
 ENDTEXT
-MENU LABEL Harvey OS
+MENU LABEL Harvey OS (CPU)
 KERNEL mboot.c32
-APPEND ../harvey
+APPEND ../harvey service=cpu nobootprompt=tcp maxcores=1024 fs=10.0.2.2 auth=10.0.2.2 nvram=/boot/nvram nvrlen=512 nvroff=0 acpiirq=1 mouseport=ps2 vgasize=1024x768x24 monitor=vesa"
```
Comment (rminnich) [inline util/syslinux-bios/syslinux/syslinux.cfg]: you really don't want to hardware fs and auth here long term, I guess it's ok for now.

Comment (gmacd) [reply util/syslinux-bios/syslinux/syslinux.cfg]: Agreed, need a better way to do this, but I think it's fine for now

File: util/img/venti.conf [hunk 4]
```diff
@@ -0,0 +1,4 @@
+index main
+isect /dev/sdE0/isect
```
Comment (rminnich) [inline util/img/venti.conf]: consider /dev/sd??/isect

Comment (rminnich) [reply util/img/venti.conf]: this is fine for now I guess

**Comment by bldybuild** [comment]
Commit 01022f1eaf4030409782caf3c84c3adb78b63b71 is not signed-off.

Please fix these issues:

>	1. ron minnich has not signed-off 01022f1.


If you'd like more information on how to sign your commits please visit [signed-off-by.me](https://signed-off-by.me)

**Comment by rminnich** [comment]
sorry, I clicked update :-(

**Comment by gmacd** [comment]
Rebased - please re-approve and we can merge it in

**Pull request merged**